test/Test/Oscoin/Protocol/Sync.hs

```diff
@@ -371,32 +382,68 @@ mockGenesis = defaultGenesis (Dict :: Dict (IsCrypto MockCrypto)) mempty
   Generators
 ------------------------------------------------------------------------------}
 
+-- Generating unique ports
+--
+-- In order to test the Sync logic for the 'RealWorld' implementation, we need
+-- to spin up some threads to actually bind on some ports in order to respond
+-- to HTTP requests from other peers. However, generating such random ports is
+-- not useful for these kind of tests, as ports on a local machine are a finite
+-- resource and generating them randomly means it's more likely to get a
+-- collision. If that happens, an exception would be raised and the tests would
+-- fail. To avoid that, we simply keep track of a global HTTP port \"dispenser\",
+-- which allocates ports sequentally and thus allows tests (even if ran in
+-- parallel) to all succeed.
+
+-- | A global counter for the HTTP ports.
+--
+-- N.B. We start from 0 for a simple reason: in case we reach the maximum
+-- number of ports (i.e. maxBound :: Word16, i.e. 65535), the counter would
+-- overflow, starting from 0. However, we cannot bind to 0 (nor to a bunch of
+-- system-used ports). This is why in `nextUnusedHttpPort' we increment the
+-- counter by 1 but we return /port + 3000/: This ensure that in case of
+-- overflow we would resume back from an allowed range.
```

Comment:
Clever :+1:

Reply:
@cloudhead dude, go away! You shouldn't be coding (more so working) during a public holiday! 😉 😝